src/components/TheHeader.vue

```diff
@@ -43,7 +54,28 @@
 import { getPageLinks } from '@/composables/navLink.js';
 import TheUserMenu from './TheUserMenu.vue';
 import { useDisplay } from 'vuetify';
+import { useRoute } from 'vue-router';
+import { ref, toRefs, watch } from 'vue';
+
+const props = defineProps({
+  theme:{
+      type:String,
+      default:'light'
+    }
+});
 
+const route = useRoute();
+const { theme } = toRefs(props);
 const { mobile } = useDisplay();
 const pageLinks = getPageLinks();
+const linkColor = ref(theme.value === 'light' ? 'text-surface-1' : 'text-surface-2');
```

Comment:
If you compute this, using the theme/variant, then we won't need the watcher. Generally favor computed props over watchers whenever possible
```suggestion
const linkColor = computed(() => variant.value === 'light' ? 'text-surface-1' : 'text-surface-2');
```